### PR TITLE
Mission.sqm changes for Contact

### PR DESCRIPTION
**Pull request by NikkoJT**
Adds template units for Contact factions (NPR, LDF). Also increments the version counter in the mission title. Template units provided by @costno 

Fixes the remaining checkbox in #254

### CODE REVIEW

**Comment by NikkoJT** [comment]
> * `GrpNATO_COV` may have lost its call sign? (it was `NATO COV -`) We should be able to catch this error also when viewing the ORBAT ingame I believe. (I didn't check if any of the new groups are correct)

I had to re-add all the editor callsigns because Costno broke them while doing the template units. Perhaps I missed one.

> * briefingName has been updated for 3.5.4

This is in the PR description. Thought I'd do it while I was making mission.sqm changes.

> * `Alpha/Bravo/Juliet` call signs changed to `A/B/J`?

See COV. I couldn't remember what they were before so I used A/B/C/J. Doesn't really matter as they'll be overriden by the groupID component anyway.

> * Some `IFV` are now `TECH`

This is accurate to the F3 groupID callsigns for these units. For example:
> 	["GrpFIA_IFV1",    _ifv, "TECH1",  "ColorRed",     "FIA TECH1 -"],

**Comment by shadow-fa** [comment]
`I used A/B/C/J`
This is now (probably) inconsistent with the groupID component <https://github.com/folkarps/F3/blob/607be67a1761d17398bcea8a14afe43210371c59/f/groupMarkers/fn_groupData.sqf#L160-L163>
Which brings me to my main question(s):
Why do we actually need the in-editor call signs? Do groupID callsigns not always work? When do they not work? During briefing? For JIP?

**Comment by NikkoJT** [comment]
> `I used A/B/C/J`
> This is now (probably) inconsistent with the groupID component

It's fixed in the commit I just made.

> Which brings me to my main question(s):
> Why do we actually need the in-editor call signs? Do groupID callsigns not always work? When do they not work? During briefing? For JIP?

groupID callsigns work in every in-game case I've seen. Editor callsigns are not reliable in multiplayer.
What they _are_ useful for is when you're building the mission in the editor. Otherwise everything is just "Alpha 1-3" etc and you have to expand or open attributes to know which group you're clicking on.

**Comment by shadow-fa** [comment]
Incrementing the version was usually a seperate pr/commit which happens once was merged and tested. but I suppose it's also okay here. See also <https://github.com/folkarps/F3/wiki/Release-Checklist> where we aim to document this process.

Since we already did that change for `mission.sqm` here, it might also make sense to do the same for `description.ext` and `README.md` in the same branch/PR such that the numbers are consistent (assuming we squash the pr commits)? ( @SamLex )

File: mission.sqm [hunk 1]
```diff
@@ -44562,7 +44562,7 @@ class Mission
 									"STRING"
 								};
 							};
-							value="Gendarmerie Juliett -";
+							value="Gendarmerie Juliet -";
```
Comment (shadow-fa) [inline mission.sqm]: Huh, it's actually `Juliett` (according to the NATO phonetic alphabet) and we consistently use it wrong throughout F3, aren't we?

**Comment by shadow-fa** [comment]
I see one discrepancy related to IFV/TECH groups (which might also affects groupData stuff):

For Syndikat, the map marker is `TECH` where as their call sign is `IFV`. 
I think this should be changed to `TECH` in `mission.sqm` and `fn_groupData.sqf`.

(All 3 FIA and 3IFB use `TECH` for their callsign and map marker text. All other groups use `IFV` for both. Which seems correct.)

**Comment by shadow-fa** [comment]
Things I noticed (which, again, might not be entirely due to contact changes):

1. Why do the following units specifically have `spetsnaz` in their call to assignGear in their init fields?

```
UnitNPR_ASL_SL
UnitNPR_BSL_SL
UnitNPR_CO
UnitNPR_CSL_SL
```
their init field looks like this:
```sqf
[this,"npr"] call f_fnc_setVirtualFaction; ["dc",this,"spetsnaz"] call f_fnc_assignGear;
```
but shouldn't it be the following?
```sqf
[this,"npr"] call f_fnc_setVirtualFaction; ["dc",this] call f_fnc_assignGear;
```
Is there any reason for this that I'm not aware of?

2. This additional field (`"spetsnaz"` and whatever the name for 3IFB is) however is missing for the NPR/3IFB crates, which do in fact need it: CrateNPR_A, CrateNPR_B, CrateNPR_C, Crate3IFB_A, Crate3IFB_B, Crate3IFB_C

3. Why do the following units have `ctrg` in that same field? It should be empty I believe.
```
UnitFIA_BSL_SL
UnitFIA_CO
UnitFIA_CO_JTAC
```

4. I noticed that `VehSyn_IFV1` and `VehSyn_IFV2` have `"ind_c_f"` for that field, while no other Syndikat vehicles have it (Apart from `VehSyn_TH1` which uses `"syndikat"`)
This is because these 3 vehicles are not actual syndikat vehicles. 
Nothing is wrong here, but it might be confusing, because we only do this for some vehicles and with different faction names.
(For 3IFB and NPR vehicles we consistently use `f_fnc_setVirtualFaction`).

**Comment by NikkoJT** [comment]
> Things I noticed (which, again, might not be entirely due to contact changes):
> 
> 1. Why do the following units specifically have `spetsnaz` in their call to assignGear in their init fields?
> [this,"npr"] call f_fnc_setVirtualFaction; ["dc",this,"spetsnaz"] call f_fnc_assignGear;
> ```
> 
> but shouldn't it be the following?
> 
> ```sqf
> [this,"npr"] call f_fnc_setVirtualFaction; ["dc",this] call f_fnc_assignGear;
> ```
> 
> Is there any reason for this that I'm not aware of?

This, and the FIA CTRG units, are because certain leadership positions in the template groups are considered to be "advisor" units. For example, the FIA commander is in fact described as a CTRG commander and receives CTRG gear. The FIA base faction is retained for purposes of markers and callsigns. This has been the case for FIA since...pretty much forever, and it's been carried forward to NPR/Spetsnaz. In other words, it's supposed to be like that.

> 1. This additional field (`"spetsnaz"` or `"npr"`??? and whatever the name for 3IFB is) however is missing for the NPR/3IFB crates, which do in fact need it: CrateNPR_A, CrateNPR_B, CrateNPR_C, Crate3IFB_A, Crate3IFB_B, Crate3IFB_C

Are you sure they need it? They've been virtualfactionned so should receive the crate loadouts appropriate to their virtual faction.

> 1. I noticed that `VehSyn_IFV1` and `VehSyn_IFV2` have `"ind_c_f"` for that field, while no other Syndikat vehicles have it (Apart from `VehSyn_TH1` which uses `"syndikat"`)
>    This is because these 3 vehicles are not actual syndikat vehicles.
>    Nothing is wrong here, but it might be confusing, because we only do this for some vehicles and with different faction names.
>    (For 3IFB and NPR vehicles we consistently use `f_fnc_setVirtualFaction`).

This is because Syndikat was done before the introduction of Virtual Faction. I don't see any particular need to change it, as it works fine and the method is documented. Also, this has nothing to do with Contact changes.

**Comment by shadow-fa** [comment]
> Are you sure they need it? They've been virtualfactionned so should receive the crate loadouts appropriate to their virtual faction.

I didn't check all of them but Syndikat uses FIA crates and has the wrong gear in it. Others might be fine if they're using virtual faction.

**Comment by NikkoJT** [comment]
> I didn't check all of them but Syndikat uses FIA crates and has the wrong gear in it. Others might be fine if they're using virtual faction.

Syndikat crates have FIA equipment in them in the template because that is the default contents for a FIA crate. However, as soon as the mission actually starts, all that gear will be removed and replaced by their assignGear loadout, which uses the `ind_c_f` (Syndikat) override parameter. Remember, how inventories appear in mission.sqm is what they look like _before_ assignGear is applied.

Unless you've started the mission, looked in the crate inventory _then_, and found the contents to still be incorrect - this is not an issue.

**Comment by shadow-fa** [comment]
I did test this ingame and assumed that the gear i saw in the crate was wrong since they were all very different weapons/magazines to the couple of syndikat units i have tested. e.g. many have AK-12 with 7.62mm 30rnd AK12 Mag, but this isn't in the crate. I checked the assignGear files and they should be there though. So something isn't working properly?

I just checked again. The crate content is the same as LDF, which is also wrong.
NPR and NATO worked just fine.

This might also not be related to the contact update sorry, but it was wrong when I tested this.
I'll try to figure out what the problem is and create a seperate issue/pr for it.

Can you confirm that this is an issue, maybe it's just me?

**Comment by SamLex** [comment]
For clarity, things changed in this PR are?:
- Added NPR and LDF to template
- Changed version in mission.sqm to 3.5.4
- AND???

**Comment by NikkoJT** [comment]
> For clarity @NikkoJT @shadow-fa, things changed in this PR are?:
> 
> * Added NPR and LDF to template
> * Changed version in mission.sqm to 3.5.4
> * AND???

Changed the Syndikat TECH1/2 callsign in groupData to actually be TECH1/2 (it was incorrectly IFV1/2).

**Pull request merged**